Cabal/doc/developing-packages.rst

```diff
@@ -1735,12 +1741,20 @@ system-dependent values for these fields.
     an executable in the package will override any of the hard-coded packages with the same name,
     and a hard-coded package will override any executable on the ``PATH``.
 
-    In the first two cases, the list entry is desguared into a :pkg-field:`build-tool-depends` entry.
+    In the first two cases, the list entry is desugared into a :pkg-field:`build-tool-depends` entry.
     In the first case, the entry is desugared into a :pkg-field:`build-tool-depends` entry by prefixing with ``$pkg:``.
     In the second case, it is desugared by looking up the package and executable name in a hard-coded table.
     In either case, the optional version bound is passed through unchanged.
     Refer to the documentation for :pkg-field:`build-tool-depends` to understand the desugared field's meaning, along with restrictions on version bounds.
 
+    If backwards-compatibility is important, you may opt to use
```

Comment:
Erm, perhaps I didn't explain clearly. Let's suppose that you're using an external program via `{-# OPTIONS_GHC -F -pgmF my-custom-preprocessor #-}`. GHC relies on being able to invoke `my-custom-preprocessor` from the PATH it's invoked with. So how do we get it into the PATH?

In old Cabal, cabal-install won't do this for you unless you `cabal install my-custom-preprocessor`. One way to cause this command to be run is if you `build-depends: my-custom-preprocessor` and you have a dummy library in the package; then it will get installed. Or you can ask the user to install it manually.